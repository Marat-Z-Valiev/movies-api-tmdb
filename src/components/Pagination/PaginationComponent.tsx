import React from "react";
import Pagination from "react-js-pagination";
import {PaginationStyled, PaginationBlockStyled} from "./PaginationStyled";
import PropTypes from "prop-types";

interface PaginationProps {
	currentPage: number;
	resultsPerPage: number;
	totalResults: any;
	handlePageChange: any;
}

const PaginationComponent: React.FC<PaginationProps> = ({
	currentPage,
	resultsPerPage,
	totalResults,
	handlePageChange,
}) => {
	let windowWidth = window.innerWidth;
	return (
		<PaginationBlockStyled>
			<PaginationStyled>
				<Pagination
					activeClass="active"
					pageRangeDisplayed={windowWidth > 425 ? 10 : 5}
					activePage={currentPage}
					itemsCountPerPage={resultsPerPage}
					totalItemsCount={parseInt(totalResults)}
					onChange={handlePageChange}
				/>
			</PaginationStyled>
		</PaginationBlockStyled>
	);
};

PaginationComponent.propTypes = {
	currentPage: PropTypes.number.isRequired,
	resultsPerPage: PropTypes.number.isRequired,
	totalResults: PropTypes.string.isRequired,
	handlePageChange: PropTypes.func.isRequired,
};

export default PaginationComponent;
