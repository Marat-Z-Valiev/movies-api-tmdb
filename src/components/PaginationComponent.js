import React from "react";
import Pagination from "react-js-pagination";
import {
	PaginationStyled,
	PaginationBlockStyled
} from "./styles/PaginationStyled";
import PropTypes from "prop-types";

const PaginationComponent = ({
	currentPage,
	resultsPerPage,
	totalResults,
	handlePageChange
}) => {
	let windowWidth = window.innerWidth;
	return (
		<PaginationBlockStyled>
			<PaginationStyled>
				<Pagination
					activeClass="active"
					pageRangeDisplayed={windowWidth > 425 ? 10 : 5}
					activePage={currentPage}
					itemsCountPerPage={resultsPerPage}
					totalItemsCount={parseInt(totalResults)}
					onChange={handlePageChange}
				/>
			</PaginationStyled>
		</PaginationBlockStyled>
	);
};

PaginationComponent.propTypes = {
	currentPage: PropTypes.number.isRequired,
	resultsPerPage: PropTypes.number.isRequired,
	totalResults: PropTypes.number.isRequired,
	handlePageChange: PropTypes.func.isRequired
};

export default PaginationComponent;
