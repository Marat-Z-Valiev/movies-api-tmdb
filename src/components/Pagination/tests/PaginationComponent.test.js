import React from "react";
import {shallow, configure} from "enzyme";
import toJSON from "enzyme-to-json";
import Adapter from "enzyme-adapter-react-16";
import PaginationComponent from "../PaginationComponent";

configure({adapter: new Adapter()});

describe("PaginationComponent", () => {
	it("should render correctly", () => {
		const wrapper = shallow(
			<PaginationComponent
				currentPage={1}
				resultsPerPage={10}
				totalResults={"100"}
				handlePageChange={() => {}}
			/>
		);
		expect(toJSON(wrapper)).toMatchSnapshot();
	});
});
