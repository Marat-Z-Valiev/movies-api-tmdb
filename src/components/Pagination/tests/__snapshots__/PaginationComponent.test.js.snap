// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationComponent should render correctly 1`] = `
<PaginationStyled__PaginationBlockStyled>
  <PaginationStyled>
    <Pagination
      activeClass="active"
      activePage={1}
      firstPageText="«"
      getPageUrl={[Function]}
      hideFirstLastPages={false}
      innerClass="pagination"
      itemsCountPerPage={10}
      lastPageText="»"
      nextPageText="⟩"
      onChange={[Function]}
      pageRangeDisplayed={10}
      prevPageText="⟨"
      totalItemsCount={100}
    />
  </PaginationStyled>
</PaginationStyled__PaginationBlockStyled>
`;
